### ISSUE TITLE
[Feature Request] Search Plex before sending the request to Radarr/Sonarr

### ISSUE THREAD

**Issue by chazlarson** [darkalfx/requestrr#47]
I've got a Plex library that is pointing at movies/tv which are not managed by Radarr/Sonarr.

I'd like Requestrr to search Plex before sending the request on to Radarr/Sonarr.  It could then find my existing copy of "Cool Space Movie" before sending it to Radarr, which would ultimately download a second copy or never tell my friend that it was available, even though it has been all along.

**Comment by LilPeen** [comment]
+1 to this, I jumped over to this bot from Mellow as it seems it is no longer being worked on, love how simple this is to setup by comparison but as I have a handful of users that request without checking plex themselves, Mellow bot will still need to be in use right now >.<

**Comment by darkalfx** [comment]
Requestrr checks if the content is available in Radarr/Sonarr before allowing users to request it.

Therefore, it wont allow users to request content that is flagged as downloaded in Sonarr/Radarr.

**Comment by LilPeen** [comment]
I can't speak for everyone, but I for one had a large library before using radarr/sonarr or even starting this venture. It's quite a pain to update radarr/sonarr with all the content I've had outside of it. thus why I second this request. I'm sure I'm asking alot but I'm sure the ones who would love this are in the same boat as Chaz/I.

**Comment by chazlarson** [comment]
Exactly.  I have a library of 30,000 movies in Plex that is stored in team drives I don’t control.  I’m not going to add all those to Radarr, so as far as Requestrr is concerned they don’t exist.

**Comment by bakes82** [comment]
@darkalfx 
A better solution would be to check a collection in the Trakt API, otherwise you now have to check Plex, the next guy wants Emby, then JellyFin.

Centralize it to Trakt, you push your collection to Trakt and boom now use any media server you want.

**Comment by LilPeen** [comment]
> @darkalfx
> A better solution would be to check a collection in the Trakt API, otherwise you now have to check Plex, the next guy wants Emby, then JellyFin.
> 
> Centralize it to Trakt, you push your collection to Trakt and boom now use any media server you want.
> 
> Emby has this option:
> ![image](https://user-images.githubusercontent.com/3197191/94951287-0042ce00-04b2-11eb-8a6c-3626ac1fa304.png)
> 
> I also belive the plex trakt plugin has the same one. All movies/tv shows get added in to trakt and marked as you having it it, and the watched history is another area

This doesn't work in Plex's case, they want you to have Trakt Premium... and who want's to pay to just track their library..

**Comment by bakes82** [comment]
First, why are you even using plex to add stuff to the collection, the bot can do it with the trakt api it self.  Also if you have the trakt plugin in plex you can do the same thing.

![image](https://user-images.githubusercontent.com/3197191/99089320-1fe21180-259b-11eb-980e-bd7f87aa213d.png)

**Comment by chazlarson** [comment]
For my part, I'm not using Plex to "add things to the collection".  I'm pointing Plex at a folder full of movies/shows that I do not have in Radarr.  So, for example, "Metropolis" is available in Plex, but it's not in Radarr.  I don't want or need a second copy of it, I have that one available to Plex.

If someone uses Requestrr to request "Metropolis", it will get added to Radarr, downloaded again, and now there are two in Plex.  I'd rather that not happen.  I'd rather that user get told "It's already available, enjoy!"

Currently the search that Requestrr does is "Is this thing available in Radarr/Sonarr?".  This request is to make that search instead be "Is this thing available in the media collection?" where the media collection is "whatever's in Radarr/Sonarr" + "whatever's in some list".  

It's a good point that "some list" shouldn't be just Plex; a Trakt list would be a fine alternative.

**Comment by LilPeen** [comment]
The api alone is free? I checked the site and even signed up but ran into this:

![image](https://user-images.githubusercontent.com/28698016/99115981-4dda4c80-25c1-11eb-97f9-7224a8b6b227.png)

**Comment by bakes82** [comment]
Are you a developer?  Yes the api is free.  You keep pointing to built in things.  Everything Im talking about needs to be coded.  The bot would write to a trakt collection not radarr not sonarr not plex or emby.  I dont know what you are trying to do lol.  The scrobbler is only to record watch status.


https://trakt.docs.apiary.io/#reference/users/add-list-items

**Comment by CoffeeKnyte** [comment]
I'd like to put my vote in for this as well. 
I love requestrr, but I also have a ton of movies/shows that weren't added through sonarr/radarr.
It would be great to have the option to search plex for that content before adding to radarr/sonarr.
I understand that plex will sometimes have a mismatch, but it's relatively rare, and the same mismatch issue can happen with radarr/sonarr as well.

Hope that this feature can be added.

**Comment by darkalfx** [comment]
You can integrate Requestrr with other tools like Overseerr, those other programs will check with Plex and provide better accuracy with your results if you don't keep all your downloads visible for Sonarr/Radarr.
